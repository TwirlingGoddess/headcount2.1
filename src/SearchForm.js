import React, { Component } from 'react';
import PropTypes from 'prop-types'

class SearchForm extends Component {
	constructor() {
		super()

		this.state = {
			value: ''
		}
	}

	handleChange = (event) => {
		this.setState({
			value: event.target.value
		})
	}

	render() {
		return(
			<form>
				<input 	type='text' 
						value={this.state.value} 
						placeholder="Type in district to Search"
						onChange={this.handleChange}
				/>
			</form>
		)
	}
}

export default SearchForm