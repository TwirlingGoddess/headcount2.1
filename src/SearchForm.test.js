import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let handleMock;
  let wrapper;
  let mockString
  let expected

  beforeEach(() => {
    handleMock = jest.fn()
    wrapper = shallow(<SearchForm onChange={handleMock} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

	it('should have a default state of an empty string', () => {
		expected = ''
		expect(wrapper.state().searchWord).toEqual(expected)
	})

  it('updates the state of when the input value changes', () => {
		mockString = { target: { value: 'abc'} }
	  expected = 'abc' ;
	  wrapper.instance().handleChange(mockString)
	  expect(wrapper.state().searchWord).toEqual(expected)
  })

  it('should invoke handleChange function when input value changes', () => {
		wrapper = mount(<SearchForm onChange={handleMock} />)
		const spy = spyOn(wrapper.instance(), 'handleChange')
    wrapper.instance().forceUpdate()
		mockString = { target: { value: 'taco'} }
		wrapper.find('.Search').simulate('change', mockString)
    expect(spy).toHaveBeenCalled()
  })




})