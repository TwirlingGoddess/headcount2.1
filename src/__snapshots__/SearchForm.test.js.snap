// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    onChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="Search"
        onChange={[Function]}
        placeholder="Type in district to Search"
        type="text"
        value={undefined}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Search",
        "onChange": [Function],
        "placeholder": "Type in district to Search",
        "type": "text",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="Search"
          onChange={[Function]}
          placeholder="Type in district to Search"
          type="text"
          value={undefined}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Search",
          "onChange": [Function],
          "placeholder": "Type in district to Search",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
